### PR TITLE
Autoimports on unresolved references

### PR DESCRIPTION
**Pull request by PolinB**

### CODE REVIEW

File: src/main/kotlin/com/compiler/server/compiler/components/ErrorAnalyzer.kt [hunk 1]
```diff
@@ -44,18 +45,24 @@ import java.util.ArrayList
...
+  private val indexationProvider: IndexationProvider
+) {
+  private val UNRESOLVED_REFERENCE_PREFIX = "Unresolved reference: "
 
   fun errorsFrom(
     files: List<KtFile>,
     coreEnvironment: KotlinCoreEnvironment,
-    isJs: Boolean = false
+    isJs: Boolean = false,
+    withImports: Boolean = false
```
Comment (AlexanderPrendota) [inline src/main/kotlin/com/compiler/server/compiler/components/ErrorAnalyzer.kt:58]: when we don't need the imports?

File: src/main/kotlin/com/compiler/server/compiler/components/ErrorAnalyzer.kt [hunk 2]
```diff
@@ -223,6 +247,15 @@ class ErrorAnalyzer(private val kotlinEnvironment: KotlinEnvironment) {
       .groupBy { it.first }
       .map { it.key to it.value.fold(emptyList<ErrorDescriptor>()) { acc, (_, errors) -> acc + errors } }
       .toMap()
+
+  private fun completionsForErrorErrorMessage(message: String, withImports: Boolean): List<Completion>? {
```
Comment (AlexanderPrendota) [inline src/main/kotlin/com/compiler/server/compiler/components/ErrorAnalyzer.kt:251]: `ErrorError`

File: src/main/kotlin/com/compiler/server/compiler/components/IndexationProvider.kt [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import common.model.ImportInfo
+import org.apache.commons.logging.LogFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Component
+import java.io.File
+import javax.annotation.PostConstruct
+
+@Component
+class IndexationProvider {
+  @Value("\${indexes.file}") private val indexesFileName: String = ""
```
Comment (AlexanderPrendota) [inline src/main/kotlin/com/compiler/server/compiler/components/IndexationProvider.kt:16]: in constructor

File: src/main/kotlin/com/compiler/server/compiler/components/IndexationProvider.kt [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+  private var ALL_INDEXES: List<ImportInfo>? = null
+  private val log = LogFactory.getLog(KotlinProjectExecutor::class.java)
+  private val UNRESOLVED_REFERENCE_PREFIX = "Unresolved reference: "
+
+  @PostConstruct
+  private fun initIndexes() {
+    println("INIT")
```
Comment (AlexanderPrendota) [inline src/main/kotlin/com/compiler/server/compiler/components/IndexationProvider.kt:24]: remove

File: src/main/kotlin/com/compiler/server/controllers/CompilerRestController.kt [hunk 5]
```diff
@@ -26,7 +26,7 @@ class CompilerRestController(private val kotlinProjectExecutor: KotlinProjectExe
   ) = kotlinProjectExecutor.complete(project, line, ch)
 
   @PostMapping("/highlight")
-  fun highlightEndpoint(@RequestBody project: Project): Map<String, List<ErrorDescriptor>> = kotlinProjectExecutor.highlight(project)
+  fun highlightEndpoint(@RequestBody project: Project) = kotlinProjectExecutor.highlight(project)
```
Comment (AlexanderPrendota) [inline src/main/kotlin/com/compiler/server/controllers/CompilerRestController.kt:29]: return type

**Review by AlexanderPrendota — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/compiler/server/compiler/components/ErrorAnalyzer.kt (hunk 1 above)
Comment (PolinB) [reply src/main/kotlin/com/compiler/server/compiler/components/ErrorAnalyzer.kt:58]: 1. In translate(files: List<KtFile>, arguments: List<String>, coreEnvironment: KotlinCoreEnvironment)
2. In completion when we check that prefix unresolved or not. We don't want imports for all unresolved references.

**Review by AlexanderPrendota — APPROVED** [review]

File: src/main/kotlin/com/compiler/server/compiler/components/IndexationProvider.kt [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+import java.io.File
+import javax.annotation.PostConstruct
+
+@Component
+class IndexationProvider(
+  @Value("\${indexes.file}") private val indexesFileName: String
+) {
+  private var ALL_INDEXES: List<ImportInfo>? = null
+  private val log = LogFactory.getLog(KotlinProjectExecutor::class.java)
+  private val UNRESOLVED_REFERENCE_PREFIX = "Unresolved reference: "
```
Comment (AlexanderPrendota) [inline src/main/kotlin/com/compiler/server/compiler/components/IndexationProvider.kt:20]: I have already seen that constant in ErrorAnalyzer

**Review by AlexanderPrendota — CHANGES_REQUESTED** [review]

**Review by AlexanderPrendota — APPROVED** [review]

**Pull request merged**